lib/yaml.rb

```diff
@@ -2,7 +2,12 @@
 
 begin
   require 'psych'
-rescue LoadError
+rescue LoadError => ex
+  # Skip warning message concerning missing yaml dependency as LoadError wasn't
+  # triggered due to a missing dependency but because of a conflicting version
+  # which was already activated.
+  raise if ex.message.match?(/already activated/)
```

Comment:
RubyGems raises this as `Gem::LoadError` which inherits from `::LoadEror` so perhaps better to check the actual exception class, which is more unlikely to change that the exception message.